python/lsst/ts/externalscripts/make_love_stress_tests.py

```
@@ -0,0 +1,318 @@
+                of messages to store before calculating the mean latency
+            * data : a list, where each element is a string in the form:
+
+                * CSC name and optional index as ``csc_name:index`` (a `str`).
+
+        Notes
+        -----
+        Saves the results on several attributes:
+
+        * location : a string, with the location of the LOVE instance to stress
+        * number_of_clients : a number, with the number of clients to create
+        * number_of_messages : a number, with the number of messages to store
+        * remotes : a dict of (csc_name, index): remote,
+            an `lsst.ts.salobj.Remote`
+
+        Constructing a `salobj.Remote` is slow (DM-17904), so configuration
+        may take a 10s or 100s of seconds per CSC.
+        """
+        self.log.info("Configure started")
+
+        # set configurations
+        self.location = config.location
+        self.number_of_clients = config.number_of_clients
+        self.number_of_messages = config.number_of_messages
+
+        # get credentials
+        self.username = "user"
+        self.password = os.environ.get("USER_USER_PASS")
+
+        if self.password is None:
+            raise Exception("Configuration failed")
+
+        # construct remotes
+        remotes = dict()
+        for elt in config.data:
+            name, index = salobj.name_to_name_index(elt)
+            self.log.debug(f"Create remote {name}:{index}")
+            if elt not in remotes:
+                remote = salobj.Remote(domain=self.domain, name=name, index=index)
+                remotes[elt] = remote
+
+        self.remotes = remotes
+
+    async def run(self):
+        """Run script."""
+        tasks = [
+            remote.start_task
+            for remote in self.remotes.values()
+            if not remote.start_task.done()
+        ]
+        if tasks:
+            self.log.info(f"Waiting for {len(tasks)} remotes to be ready")
+            await asyncio.gather(*tasks)
```

Will you ever have no remotes? What does the script do in that situation? Should it fail?

---

I think this shouldn't happen as the `config.data` param should always have at least 1 item, defined by the schema:
```
data:
    description: List of CSC_name[:index]
    type: array
    minItems: 1
    items:
```

---

Then why the `if tasks`? I suggest omitting that if statement and always logging and awaiting

---

I was looking at this and there is case which will have `tasks` as an empty array. As the array is fille with a conditional `if not remote.start_task.done()`, in the case where all remotes start_task param future is done, then the array will be empty.

---

Fair enough.

It would simpler and more robust to accumulate all of them and await all of them:
```
self.log.info(f"Waiting for {len(self.remotes)} remotes to be ready")
await asyncio.gather(*[remote.start_task for remote in self.remotes.values()])
```
It is more robust because if one of the remote.start_task has already failed, this gather will raise an exception, whereas your current code silently ignores such a failure.